### PR TITLE
WIP / PoC of `SpecialEmptyMatMaterializerSpec` for Play

### PR DESCRIPTION
**Pull request by ktoso**
Instead of special casing when to handle what, we can provide a materializer that assumes "aha, no data will flow here".

This is a very rough sketch and WIP branch, not ready for proper review yet.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+    val interpreter = new ActorGraphInterpreterLogic {
+      override val _initial: GraphInterpreterShell = shell
+      override def log: LoggingAdapter = NoLogging
+      override def self: ActorRef = null
+      override def context: ActorContext = ???
+    }
+
+    interpreter.preStart()
+    interpreter.processEvent(shell.resume)
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala]: FIXME also do postStop etc

File: akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+    shell.connections = finalConnections
+    shell.logics = logics.toArray(logicArrayType)
+
+    val interpreter = new ActorGraphInterpreterLogic {
+      override val _initial: GraphInterpreterShell = shell
+      override def log: LoggingAdapter = NoLogging
+      override def self: ActorRef = null
+      override def context: ActorContext = ???
+    }
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala]: very hacky

File: akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+  val shell = new GraphInterpreterShell(
+    connections = null,
+    logics = null,
+    effectiveSettings,
+    materializer)
+
+  override def name: String = "Fusing GraphStages phase"
+
+  override def materializeAtomic(mod: AtomicModule[Shape, Any], attributes: Attributes): (GraphStageLogic, Any) = {
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala]: TODO clean up, only small bits are changed

File: akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+    val matAndLogic = stage.createLogicAndMaterializedValue(attributes)
+    val logic = matAndLogic._1
+    logic.originalStage = OptionVal.Some(stage)
+    logic.attributes = attributes
+    logics.add(logic)
+    logic.stageId = logics.size() - 1
+
+    fullIslandName match {
+      case OptionVal.Some(_) ⇒ // already set
+      case OptionVal.None    ⇒ fullIslandName = OptionVal.Some("FAST_NAME")
```
Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/SpecialEmptyMaterializer.scala]: Give better names, perhaps we can get away without the full SeqNames though

**Comment by patriknw** [comment]
@ktoso Shall we stash and close this one?

**Comment by ktoso** [comment]
Yes, closing - it was not needed in the end since we managed to get away with sub fusing

**Pull request closed without merging**